### PR TITLE
Upmerge 14.12.2020

### PR DESCRIPTION
**Pull request by de-nordic**
Updated revision of zephyr-sdk.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Comment by jhn-nordic** [comment]
applications/connectivity_bridge has issues. CONFIG_SYS_PM_POLICY_APP has to be renamed to CONFIG_PM_POLICY_APP in prj.conf and uart_handler.c, but then it overflows RAM a lot. Will need to investigate

**Review by jhn-nordic — CHANGES_REQUESTED** [review]
applications/connectivity_bridge is broken (thingy91_nrf52840)

**Review by rlubos — APPROVED** [review]
Networking side looks good.

**Review by nordic-auko — APPROVED** [review]
connectivity_bridge application and alexa_gadget sample looks good

**Comment by de-nordic** [comment]
Can someone look at the nrf/tests/subsys/dfu/dfu_target_mcuboot example?

**Comment by MarekPieta** [comment]
I would like to make some manual checks for nrf_desktop before this goes in. I cannot add myself as a reviewer (that exceeds limit of 15 people).

**Comment by hakonfam** [comment]
> Can someone look at the nrf/tests/subsys/dfu/dfu_target_mcuboot example?

I will do that

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebug.conf [hunk 1]
```diff
@@ -66,7 +66,8 @@ CONFIG_HEAP_MEM_POOL_MIN_SIZE=32
 
 CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
 
-CONFIG_SYS_PM_POLICY_APP=y
+CONFIG_PM_POLICY_APP=y
+CONFIG_PM_DEEP_SLEEP_STATES=y
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebug.conf:70]: No need to select this explicitly. Power manager Kconfig option selects this one.

Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebug.conf:70]: Also for other configurations.

File: applications/nrf_desktop/src/modules/Kconfig.power_manager [hunk 2]
```diff
@@ -8,9 +8,9 @@ menu "Power manager"
 
 config DESKTOP_POWER_MANAGER_ENABLE
 	bool "Enable power management"
-	depends on HAS_SYS_POWER_STATE_DEEP_SLEEP_1
+	depends on HAS_POWER_STATE_DEEP_SLEEP_1
 	select DEVICE_POWER_MANAGEMENT
-	select SYS_POWER_DEEP_SLEEP_STATES
+	select PW_DEEP_SLEEP_STATES
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.power_manager:13]: `PM_DEEP_SLEEP_STATES`

**Review by MarekPieta — CHANGES_REQUESTED** [review]
I noticed USB related problem for nrf_desktop devices. I am investigating it:
`<err> usb_nrfx: Buffer alloc failed for EP 0x01`

Edit:
PR with fix: https://github.com/zephyrproject-rtos/zephyr/pull/30980

**Review by trond-snekvik — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]
Everything works fine for nrf_desktop application

**Review by pdunaj — APPROVED** [review]

**Comment by de-nordic** [comment]
> 
> 
> applications/connectivity_bridge is broken (thingy91_nrf52840)

@jhn-nordic Has anything changed with the app?

**Review by hakonfam — CHANGES_REQUESTED** [review]
TF-M sample is broken, looking at it

**Comment by jhn-nordic** [comment]
> > applications/connectivity_bridge is broken (thingy91_nrf52840)
> 
> @jhn-nordic Has anything changed with the app?

I believe @nordic-auko  fixed this before the holidays

**Comment by hakonfam** [comment]
> TF-M sample is broken, looking at it

@de-nordic here is a patch: https://github.com/de-nordic/sdk-zephyr/pull/2

**Review by joerchan — APPROVED** [review]

**Review by grochu — APPROVED** [review]
Looks good from the Protocols point of view. Good job!

**Pull request merged**